jsr352/core/src/main/java/org/hibernate/search/jsr352/massindexing/MassIndexingJobAvailableParameters.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.jsr352.massindexing;
+
+
+/**
+ * @author Yoann Rodiere
+ */
+public final class MassIndexingJobAvailableParameters {
```

Comment:
> There's another thing we could do, that would reduce the confusion between properties and parameters: we could only use @BatchProperty in the JobContextSetupListener, and make the listener responsible for putting every property in the JobContextData. The main advantage would be that every property would be appropriately deserialized and validated upfront, eliminating any risk of I/O error during the job execution. Maybe I should create a ticket about this?

I guess there's no need to do so.

Firstly because the difference between these 2 terms is pretty clear. Job parameters can be specified each time a job is started or restarted. Job parameters are keyword/value string pairs. The JobOperator start and restart operations support the specification of job parameters. And we can find examples in _8.8 Job XML Substitution_ > _8.8.2 Examples_ (page 53):

```xml
<property name="infile.name" value="#{jobParameters['infile.name']}" />
<property name="infile.name" value="#{jobParameters['infile.name']}?:in.txt;" />
```

Secondly, as we rely on the batch technology, we should trust them for the I/O handling. We can create a ticket after having found a failure. For deserialization, I guess we only need to deserialize custom criteria?